packages/c/sshnpd/src/main.c

```diff
@@ -231,9 +237,13 @@ int main(int argc, char **argv) {
   cJSON_AddItemToObject(ping_response_json, "supportedFeatures", supported_features);
 
   cJSON *allowed_services = cJSON_CreateArray();
+  char *buf = malloc(sizeof(char) * 1024);
   for (int i = 0; i < params.permitopen_len; i++) {
-    cJSON_AddItemToArray(allowed_services, cJSON_CreateString(params.permitopen[i]));
+    sprintf(buf, "%s:%u", params.permitopen_hosts[i], (unsigned int)params.permitopen_ports[i]);
+    cJSON_AddItemToArray(allowed_services, cJSON_CreateString(buf));
   }
+  free(buf);
```

Comment:
Since I split params.permitopen into params.permitopen_hosts and params.permitopen_ports